### ISSUE TITLE
Exhaustively using all the CPU threads

### ISSUE THREAD

**Issue by k3yavi** [dcjones/isolator#5]
Hi,

While running isolator we have observed that in the optimization and sampling stage of the tool it is exhaustively using all the threads available in the system bypassing the command line input provided using the `-p` flag.

**Comment by dcjones** [comment]
You're correct, this should be fixed. It will end up using n^n threads during quantification I believe.
